### ISSUE TITLE
new staticmap: area per grid cell

### ISSUE THREAD

**Issue by shartgring** [Deltares/hydromt#504]
### Kind of request

Adding new functionality

### Enhancement Description

Hi all!

I have a suggestion for an enhancement, especially related to hydromt_wflow, but generalizable to all models.

Currently, the area per grid cell is calculated during the initialization of a wflow model, and used to determine the horizontal runoff (as vertical processes are calculated in mm/cell). I think it would be useful to include a staticmap with the area per grid cell.
- This could be used by the wflow, SFINCS and FIAT models during initialization and save some computations as it will be the same for models in the same region with identical resolution.
- Secondly, it can also be helpful during the data analysis of the staticmaps, forcings and outputs.
- Furthermore, it can give insight into the 'distortion' of the projection before having to run the model by including it in the HydroMT workflow when working with large-scale models generated using HydroMT.

Two use cases where I missed this possible staticmap to illustrate my suggestion:
- Calculate precipitation in [m3] by multiplying the precipitation map with such an area map, and use it to calculate the runoff per subcatchment
- Convert the actual maximum leakage of a catchment (given as mm per grid cell) to m3 to compare the values to total water extraction volumes in [m3] mentioned by a mining company

As an alternative, I determined the area of a single grid cell and assumed that all cells were the same size. However, this assumption will not always hold, especially for large models far away from the equator.

**Comment by hboisgon** [comment]
Hi @shartgring, agreed! 
We already have a method to compute grid area in HydroMT core: [raster.area_grid](https://deltares.github.io/hydromt/latest/_generated/hydromt.DataArray.raster.area_grid.html#hydromt.DataArray.raster.area_grid) that computes area of each grid cell for both projected and geographic coordinates.

So then I think it's more up to hydromt-wflow to maybe add that map in the staticmaps with a new method? Wflow does not use this so could also decide to use the function from HydroMT when you need to do your calculations?
Alternatively I could also add a generic GridModel method to add the area map to self.grid? Then when we migrate hydromt-wflow to GridModel, the method will become available for it.

What do you think?

**Comment by shartgring** [comment]
Hi Hélène, thanks I didn't know that feature existed! It works perfectly, I only needed to specify the crs with hydromt.DataArray.raster.set_crs to get it working. I see now that my first calculations were off by ~5% so some accuracy is definitely gained

For me personally, it works fine this way, as I already use hydromt methods a lot in my scripts. But I agree with your last suggestion that it would be good to include it in hydromt-wflow when it gets migrated to GridModel (I already saw that this is listed https://github.com/Deltares/hydromt_wflow/issues/155) , and it will probably result in more people using this method when it will become available this way. 

So I think it is fine for now as the method already exists, and it would be good to include it in hydromt-wflow when GridModel gets implemented
